R/draws.R

```diff
@@ -1,10 +1,112 @@
+#' Must contain the outcome variable. See details.
+#' @param data_ice A `data.frame` that specifies the information related
+#' to the intercurrent events (ICE) and the imputation strategies. Its structure is described in the details.
+#' @param vars A `vars` object as generated by [set_vars()]. See details.
+#' @param method An object returned by the functions [method_bayes()],
+#' [method_approxbayes()] or [method_condmean()].
+#' It specifies the multiple imputation methodology to be used. See details.
+#'
+#' @details
+#'
+#' `draws` performs the first step of the multiple imputation (MI) procedure: the fit of the
+#' base imputation model. The goal is to estimate the parameters of interest needed
+#' for the imputation phase (i.e. the regression coefficients and the covariance matrices
+#' from a MMRM model).
+#'
+#' The function distinguishes between the following methods:
+#' - Bayesian MI based on MCMC sampling: `draws` returns the draws
+#' from the posterior distribution of the parameters using a Bayesian approach based on
+#' MCMC sampling. This method can be adopted by specifying `method` using [method_bayes()].
+#' - Approximate Bayesian MI based on bootstrap: `draws` returns
+#' the draws from the posterior distribution of the parameters using a Bayesian approach,
+#' where the sampling from the posterior distribution is simulated fitting the MMRM model
+#' on bootstrap samples of the original dataset. This method can be adopted by
+#' specifying `method` through [method_approxbayes()].
+#' - Conditional mean imputation with bootstrap re-sampling: `draws` returns the
+#' ML/REML parameter estimates from the original dataset and from `n_samples` bootstrap samples.
+#' This method can be adopted by specifying `method` through [method_condmean()] with
+#' argument `type = "bootstrap"`.
+#' - Conditional mean imputation with Jackknife re-sampling: `draws` returns the
+#' ML/REML parameter estimates from the original dataset and from the leave-one-out samples.
+#' This method can be adopted by specifying `method` through [method_condmean()] with
+#' argument `type = "jackknife"`.
+#'
+#' The argument `data` must have one row per visit per subject. This means that incomplete
+#' outcome data must be set as `NA` instead of having the related row missing. Missing values
+#' in the covariates are not allowed. If `data` is incomplete
+#' then the [expand_locf()] helper function can be used to add in any missing rows, using
+#' Last Observation Carried Forward (LOCF) imputation to impute the covariates values.
+#'
+#' The argument `data_ice` contains information about the occurrence of ICEs. It is a
+#' `data.frame` with 3 columns:
+#' - A vector of characters related to the ids of the subjects that experienced
+#'   the ICE (must be named as `vars$subjid`).
+#' - A vector of levels of `data[[vars$visit]]` including the first visit after the ICE (must be named as `vars$visit`). If
```

Comment:
I would potentially start with a more basic level description  i.e.  The visit on which the ICE occurred on, this value must be equal to one of the levels of data[[vars...

Reply:
done